crates/compiler/parse/src/expr.rs

```diff
@@ -311,8 +311,20 @@ fn expr_start<'a>(options: ExprParseOptions) -> impl Parser<'a, Loc<Expr<'a>>, E
 
 fn expr_operator_chain<'a>(options: ExprParseOptions) -> impl Parser<'a, Expr<'a>, EExpr<'a>> {
     line_min_indent(move |arena, state: State<'a>, min_indent: u32| {
-        let (_, expr, state) =
-            loc_possibly_negative_or_negated_term(options).parse(arena, state, min_indent)?;
+        let (_, expr, state) = loc_possibly_negative_or_negated_term(options)
+            .parse(arena, state, min_indent)
+            .map(|(progress, expr, state)| {
+                // If the next thing after the expression is a `!`, then it's Suffixed
+                if state.bytes().starts_with(b"!") {
+                    (
+                        progress,
+                        Loc::at(expr.region, Expr::Suffixed(arena.alloc(expr.value))),
```

Comment:
I'm really unsure about this. My concern is the that Loc in `expr` here points to just the ident and doesn't include the `!` in it. I'm not sure if that is an issue.

Reply:
If so, we'll see this in error messages when we get to testing those - the red underline either will or won't include the `!` depending on this. That might actually be what we want in some cases, but maybe not others!